core/src/main/java/kafka/server/share/SharePartitionManager.java

```diff
@@ -560,88 +588,58 @@ void maybeProcessFetchQueue() {
                         maybeCompleteInitializationWithException(sharePartitionKey, shareFetchPartitionData.future, throwable);
                         return;
                     }
-                    // Fetch messages for the partition only if it is active.
-                    int partitionMaxBytes = shareFetchPartitionData.partitionMaxBytes.getOrDefault(topicIdPartition, 0);
-                    // Add the share partition to the list of partitions to be fetched only if we can
-                    // acquire the fetch lock on it.
-                    if (sharePartition.maybeAcquireFetchLock()) {
-                        // If the share partition is already at capacity, we should not attempt to fetch.
```

Comment:
Seems the merge has changed things a bit as previous initilization error of share partition is now now considered.

Reply:
Hi @apoorvmittal10 , yes, I think we need some refactor between SharePartitionManager and DelayedShareFetch code to address the share partition initialization. I have created a JIRA https://issues.apache.org/jira/browse/KAFKA-17510 and we can address this in the future PRs since it does not affect the reliability of share groups code in AK.